packages/contracts/contracts/L1/Colosseum.sol

```diff
@@ -417,39 +661,88 @@ contract Colosseum is Initializable, Semver {
+    /**
+     * @notice Returns the challenge corresponding to the given L2 output index.
+     *
+     * @param _outputIndex Index of L2 checkpoint output.
+     *
+     * @return The status of challenge.
+     */
+    function getChallenge(uint256 _outputIndex) external view returns (Types.Challenge memory) {
+        return challenges[_outputIndex];
     }
 
-    function getChallengeInProgress() public view returns (Types.Challenge memory) {
-        Types.Challenge memory challenge = challenges[latestChallengeId];
-
-        require(challenge.turn > 0, "Colosseum: challenge not found");
+    /**
+     * @notice Returns the challenge status corresponding to the given L2 output index.
+     *
+     * @param _outputIndex Index of L2 checkpoint output.
+     *
+     * @return The status of challenge.
+     */
+    function getStatus(uint256 _outputIndex) external view returns (ChallengeStatus) {
+        Types.Challenge storage challenge = challenges[_outputIndex];
+        return _challengeStatus(challenge);
+    }
 
-        return challenge;
+    /**
+     * @notice Returns the segment length required for that turn.
+     *
+     * @param _turn The challenge turn.
+     *
+     * @return The segments length.
+     */
+    function getSegmentsLength(uint256 _turn) public view returns (uint256) {
+        require(_turn > 0, "Colosseum: invalid turn");
+        return segmentsLengths[_turn - 1];
     }
 
-    function isAbleToBisect() public view returns (bool) {
-        Types.Challenge memory challenge = getChallengeInProgress();
+    /**
+     * @notice Determines whether bisection is possible in the challenge corresponding to the given L2 output index.
+     *
+     * @param _outputIndex Index of L2 checkpoint output.
+     *
+     * @return Whether bisection is possible.
+     */
+    function isAbleToBisect(uint256 _outputIndex) public view returns (bool) {
+        Types.Challenge storage challenge = challenges[_outputIndex];
         return _isAbleToBisect(challenge.segSize, challenge.turn);
     }
 
-    function isInProgress() public view returns (bool) {
-        ChallengeStatus status = getStatusInProgress();
+    /**
+     * @notice Determines whether the challenge corresponding to the given L2 output index is in progress.
+     *
+     * @param _outputIndex Index of L2 checkpoint output.
+     *
+     * @return Whether the challenge is in progress.
+     */
+    function isInProgress(uint256 _outputIndex) public view returns (bool) {
+        Types.Challenge storage challenge = challenges[_outputIndex];
+        ChallengeStatus status = _challengeStatus(challenge);
 
-        // If the challenger turn times out, there is no motivation for the asserter
-        // to progress the challenge. The asserter only pay gas. So the challenger
-        // timeout status is considered to be closed, too.
         return status != ChallengeStatus.NONE && status != ChallengeStatus.CHALLENGER_TIMEOUT;
```

Comment:
Nice! Instead of checking it with ChallengeStatus.NONE, it seems that checking it with isInProgress would resolve the issue. It would be beneficial to have unit tests for this as well.
Plus, it seems more appropriate to use `isInProgress` when seeing error message.

```solidity
 function createChallenge(uint256 _outputIndex, bytes32[] calldata _segments) external payable {
        Types.Challenge storage challenge = challenges[_outputIndex];

        require(
            isInProgress(challenge),
            "Colosseum: the challenge for given output index is already in progress."
        );
```

Reply:
Oh I mean `isInProgress` function itself should be revised. Because validator will check if challenges related to himself are in progress, then if one of the challenges related to himself is in progress and he is asserter for the challenge and the status is `CHALLENGER_TIMEOUT`, he should close the challenge. So `isInProgress` should return true when `CHALLENGER_TIMEOUT` to enable the asserter to close the challenge.

Reply:
Also, it would be better to restore the deleted comments as they seem to be helpful with understanding.